### PR TITLE
Added examples for the common use cases when using the client

### PR DESCRIPTION
**Pull request by gabriela-yoncheva**
- Modify/Create a feature in a thing (with a certain definition)
- Modify a feature's property
- Messages request-response handling flow

Signed-off-by: Gabriela Yoncheva <fixed-term.Gabriela.Yoncheva@bosch.io>

### CODE REVIEW

**Comment by gabriela-yoncheva** [comment]
@thjaeckle Please do not merge this PR until it is approved by a member of the team.

File: examples/working_with_features.py
Comment (konstantina-gramatova) [inline examples/working_with_features.py:74]: All these commands must be performed after the client has notified it is conneced (i.e. in the on_connect hook). Currently, there is no guarantee that they will be executed properly. Please, fix it.


Another hint:
To keep the test representation "clean" - i.e. no changes after executing the example, maybe a better idea is to order the calls like - create/modify feature, create/modify a feature's propery, delete property, delete feature.

File: examples/working_with_features.py
Comment (konstantina-gramatova) [inline examples/working_with_features.py:33]: It's a good practice to name the features in PascalCase. Please, fix it.

File: examples/working_with_features.py
Comment (konstantina-gramatova) [inline examples/working_with_features.py:30]: Think about having the commonly used strings as private constants defined for the example - test namespaces/definition IDs, entity name, feature ID, etc.

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

File: examples/message_request_response_handling.py
Comment (tinchouuu) [inline examples/message_request_response_handling.py:36]: Correlation ID is not set - please, add it.

File: examples/message_request_response_handling.py
Comment (tinchouuu) [inline examples/message_request_response_handling.py:35]: Should be possible to use the short notation for the .inbox(message_subject) call here.

**Review by tinchouuu — CHANGES_REQUESTED** [review]

File: examples/message_request_response_handling.py
Comment (konstantina-gramatova) [inline examples/message_request_response_handling.py:36]: Correlation ID is missing - please, set it in the example.

File: examples/message_request_response_handling.py
Comment (konstantina-gramatova) [inline examples/message_request_response_handling.py:35]: The short notation of the .inbox(message_subject) should be possible to be used as well.

**Review by konstantina-gramatova — CHANGES_REQUESTED** [review]

**Review by konstantina-gramatova — APPROVED** [review]
LGTM

**Pull request merged**
